src/Internal/Workflow/WorkflowContext.php

```diff
@@ -481,8 +501,7 @@ public function awaitWithTimeout($interval, ...$conditions): PromiseInterface
     public function resolveConditions(): void
     {
         foreach ($this->awaits as $awaitsGroupId => $awaitsGroup) {
-            foreach ($awaitsGroup as $i => $cond) {
-                [$condition, $deferred] = $cond;
+            foreach ($awaitsGroup as $i => [$condition, $deferred]) {
                 if ($condition()) {
                     $deferred->resolve();
```

Comment:
do we actually still need it? Maybe we can try to delete whote awaitGroup without rejecting each of internal promises (they are our promises anyway).

Reply:
This shuld be tested